capstone/src/main/java/com/google/sps/servlets/LessonHandlerServlet.java

```diff
@@ -12,10 +26,53 @@
 @WebServlet("/lesson")
 public class LessonHandlerServlet extends HttpServlet {
 
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery roomResults = datastore.prepare(new Query(Room.ROOM_ENTITY_NAME));
+        Room room = null;
+
+        for(Entity entity : roomResults.asIterable()) {
+            if (new Room(entity).getRoomKey().getId() == Long.parseLong(request.getParameter("room_id"))) {
+                room = new Room(entity);
+            }
+        }
```

Comment:
When you get the room like this, you are querying for ALL rooms, then finding the right one in a for loop. That doesnt allow indexes to work their magic, so as the database grows, this would become incredibly slow. Instead you should set a query filter

Reply:
I don't know how I would check to see if the Room key is the same as the query parameter with no static variables set to "roomId" in the Room class. If you check out LoginServlet it was easy there because the User stores his Id but I didn't know if there was a special way to do this when an object doesn't have the filter specific field?

Reply:
Fixed